sous/image_mapping_test.go

```
@@ -0,0 +1,86 @@
+package sous
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/opentable/sous/util/docker_registry"
+	"github.com/samsalisbury/semv"
+	"github.com/stretchr/testify/assert"
+)
+
+type mdChan chan docker_registry.Metadata
+
+type DummyRegistryClient struct {
+	mds mdChan
+}
+
+func (drc *DummyRegistryClient) Cancel()                  {}
+func (drc *DummyRegistryClient) BecomeFoolishlyTrusting() {}
+
+func (drc *DummyRegistryClient) GetImageMetadata(in, et string) (docker_registry.Metadata, error) {
+	return <-drc.mds, nil
+}
+
+func (drc *DummyRegistryClient) LabelsForImageName(in string) (map[string]string, error) {
+	md := <-drc.mds
+	return md.Labels, nil
+}
+
+func TestRoundTrip(t *testing.T) {
+	assert := assert.New(t)
+
+	mds := make(mdChan, 10)
+	nc := NameCache{
+		&DummyRegistryClient{mds},
+		make(DockerNameLookup),
+		make(SourceNameLookup),
+	}
+
+	v, _ := semv.Parse("1.2.3")
```

Can use `semv.MustParse` to panic instead of returning error.